### DISCUSSION TITLE
How to link to a specific page? (translation agnostic)

### DISCUSSION THREAD

**Discussion by t1ll** [SpinaCMS/Spina#1364] [Q&A] [answered]
Hi everyone

First of all: I'm super happy to have found Spina. Finally a developer focused CMS that is simple and works nicely :) Thank you for all the effort that is put into this ❤️

I have a question which is was not able solve via the docs nor some code research: What is the best way to link to a specific page from my views, e.g. in a hero header where I want a fixed call to action to point to a specific page?

I'm not using the ::Text part, so my hero header mostly consist of ::Line and ::MultiLine. The call to action button in the hero is static and has a ::Line attribute for the text and should point to a specific other page from the CMS. 

I think i could use something like `Spina::Page.roots.regular_pages.live.where(id: 123).first.materialized_path` but it feels a bit wrong.

What is the best way to do this? There is no ::Link attribute that would allow a CMS user to set the link in the UI. 
I found the "name" attribute on the Page model, which would be better than the ID, but somehow "name" is only set for the homepage and I found no UI-way to set the name of a page.

Oh, and important: the link should be "language agnostic", eg. if the user is browsing the French version of the page it should also link to the French variant again.

Thanks for the help 🙏

**Comment by Bramjetten** [comment, marked as answer]
Thanks for the kind words! You're looking for custom pages. You can add multiple custom pages to your theme config under theme.config_pages. Those will all get a custom name, similar to your homepage. That way you can easily reference those kinds of pages without having to hardcode IDs.

**Reply by t1ll** [reply]
That worked! :) Thank you very much for the fast help

PS: i also saw that there actually is a PageLink part. I think that is not (yet) in the documentation, but would be a nice way to link to pages via the CMS
